lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py

```diff
@@ -868,6 +878,21 @@ def exec_command(self, command):
             stderr=subprocess.DEVNULL,
         )
 
+    def detect_init_system(self):
+        """Detect whether services are running with Docker or systemd."""
+
+        docker_running = "magmad" in self.exec_command_output("docker ps --filter 'name=magmad' --format '{{.Names}}'")
```

Comment:
is this robust against environments where docker/systemd is not installed?

Reply:
Good point! This actually fails on `magma_deb` where Docker is not installed. How's something like this?
```
    def detect_init_system(self) -> InitMode:
        """Detect whether services are running with Docker or systemd."""

        docker_running = False
        systemd_running = False

        try:
            docker_running = "magmad" in self.exec_command_output("docker ps --filter 'name=magmad' --format '{{.Names}}'")
        except:
            logging.info("Docker is not installed")

        try:
            systemd_running = self.exec_command("systemctl is-active magma@magmad") == 0
        except:
            logging.info("systemd is not installed")
        ...
```
Btw, is it possible `systemctl` would not be installed but `systemd` would be?

Reply:
`systemctl` should be present when `systemd` is installed, so I think we can safely assume its presence. In the `except` we should be specific which exceptions we want to catch (probably `CalledProcessError`).